.github/workflows/release.yml

```diff
@@ -1,211 +1,40 @@
-      - preconditions
-      - lint
-      - dependency-check
-      - tests
-      - check-version
-    runs-on: ubuntu-latest
-    if: ${{ needs.check-version.outputs.is_new_version == 'true' }}
-
-    steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-node@v4
-        with:
-          node-version: '18.x'
-          registry-url: 'https://registry.npmjs.org'
-          scope: '@digicatapult'
-      - name: Install packages
-        run: npm ci
-      - name: Build
-        run: npm run build
-      - name: Publish to npmjs packages
-        run: npm publish --access public
-        env:
-          NODE_AUTH_TOKEN: ${{ secrets.NPMJS_TOKEN }}
-
-  publish:
-    name: 'Publish release'
-    needs: [preconditions, lint, dependency-check, tests, check-version]
-    runs-on: ubuntu-latest
-    if: ${{ needs.check-version.outputs.is_new_version == 'true' }}
-
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: Build release version
-        uses: softprops/action-gh-release@v2
-        with:
-          token: '${{ secrets.GITHUB_TOKEN }}'
-          tag_name: ${{ needs.check-version.outputs.version }}
-          prerelease: false
-          name: ${{ needs.check-version.outputs.version }}
-          generate_release_notes: true
-      - name: Delete release latest
-        uses: actions/github-script@v7
-        with:
-          github-token: ${{secrets.GITHUB_TOKEN}}
-          script: |
-            const { owner, repo } = context.repo
-            try {
-              await github.rest.git.deleteRef({ owner, repo, ref: 'tags/latest' })
-            }
-            catch (err) {
-              if (err.status !== 422) throw err
-            }
-      - name: Build release latest
-        uses: softprops/action-gh-release@v2
-        with:
-          token: '${{ secrets.GITHUB_TOKEN }}'
-          tag_name: latest
-          prerelease: false
-          name: Latest ${{ needs.check-version.outputs.version }}
-          generate_release_notes: true
+  static-checks-npm:
+    uses: digicatapult/shared-workflows/.github/workflows/static-checks-npm.yml@main
+    with:
+      matrix_commands: '["lint", "depcheck", "check"]'
+  tests-npm:
+    uses: digicatapult/shared-workflows/.github/workflows/tests-npm.yml@main
+    with:
+      tests: '["test", "test:unit"]'
+      docker_compose_file: ''
+  release-npm:
+    needs: [static-checks-npm, tests-npm]
+    uses: digicatapult/shared-workflows/.github/workflows/release-module-npm.yml@main
+    with:
+      npm_build: true
+    permissions:
+      contents: write
+    secrets:
+      REGISTRY_AUTH_TOKEN: ${{ secrets.NPMJS_TOKEN }}
+  release-gh-packages:
+    needs: [static-checks-npm, tests-npm]
+    uses: digicatapult/shared-workflows/.github/workflows/release-module-npm.yml@main
+    with:
+      npm_build: true
+      registry_url: 'https://npm.pkg.github.com'
+    permissions:
+      contents: write
+    secrets:
+      REGISTRY_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Not sure this will work. Might need to alter shared workflow to enable publishing to a github npm package